cordformation/src/main/kotlin/net/corda/plugins/External.kt

```diff
@@ -0,0 +1,28 @@
+package net.corda.plugins;
+
+import org.gradle.api.tasks.Input
+
+open class External {
+
+    @get:Input
+    var containerName: String = "external-service"
+
+    @get:Input
+    var servicePorts: List<Int> = listOf(8080)
+
+    @get:Input
+    var containerImage = ""
+
+    @get:Input
+    var volumes: List<Map<String, String>> = emptyList()
```

Comment:
This is tricky - I suspect it should look like this:
```kotlin
@Suppress("unchecked_cast")
@get:Input
val volumes: ListProperty<Map<String, String>> = objects.listProperty(Map::class.java) as ListProperty<Map<String, String>>
```
Again, this `ListProperty` is empty by convention and so doesn't need to be declared as `@Optional`.